pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -227,24 +229,45 @@ func init() {
 	})
 }
 
+// UUID gets the BIOS UUID via the sys interface.  This UUID is known by vsphere
+func getvmUUID() (string, error) {
+	id, err := ioutil.ReadFile(UUIDPath)
+	if err != nil {
+		return "", fmt.Errorf("error retrieving vm uuid: %s", err)
+	}
+
+	uuidstr := string(id[:])
+
+	// check the uuid starts with "VMware-"
+	if !strings.HasPrefix(uuidstr, UUIDPrefix) {
+		return "", fmt.Errorf("cannot find this VM's UUID")
```

Comment:
added uuid in the error.
`return "", fmt.Errorf("Failed to match Prefix, UUID read from the file is %v", uuidFromFile)`